CMakeLists.txt

```diff
@@ -64,7 +64,7 @@ set(PUBLIC_HEADERS
     cppcodec/detail/hex.hpp
     cppcodec/detail/stream_codec.hpp)
 
-add_library(cppcodec OBJECT ${PUBLIC_HEADERS}) # unnecessary for building, but makes headers show up in IDEs
+add_library(cppcodec ${PUBLIC_HEADERS}) # unnecessary for building, but makes headers show up in IDEs
```

Comment:
Would this also work if declared as `INTERFACE` library? I think that would be ideal given that this is what cppcodec actually is.

Reply:
Sorry, I don't know this one too, it didn't work if the OBJECT is just replaced with INTERFACE as it doesn't take source arguments. I use INTERFACE only for header only libs of mine, this one requires .cpp files too, so i don't know how that works for INTERFACE.

Reply:
If we can restrict CMake versions >= 3.12, we can leave `OBJECT` here as-is, because [since version 3.12, object libraries can be linked to with `target_link_libraries()`](https://cmake.org/cmake/help/latest/command/add_library.html#object-libraries); and with added `target_include_directories()`, this library can be consumed simply via FetchContent.

Otherwise, you may need to specify `INTERFACE` in `add_library()` and use [target_sources](https://cmake.org/cmake/help/latest/command/target_sources.html#command:target_sources) to add source files.
and `target_sources` requires CMake 3.1, you also must use `INTERFACE` in `target_include_directories()`

Reply:
Fyi, maybe we can use nlohmann/json's [CMakeLists.txt](https://github.com/nlohmann/json/blob/develop/CMakeLists.txt) as a reference impl